scripts/spin-up-new-devnet.sh

```diff
@@ -2,39 +2,67 @@
 parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
 cd "$parent_path/.."
 
+echo "Current CLUSTER_ENV is $CLUSTER_ENV"
+echo "Change this env to devnet|testnet to change cluster network"
+
 echo "Stopping demo in case anything is already running"
 ./scripts/stop-demo.sh
 
+source ./scripts/setup-envs.sh
+
 echo "Prepare devnet files"
-./scripts/reset-devnet.sh
+source ./scripts/reset-devnet.sh
 
 echo "Starting cardano node"
-docker-compose up cardano-node -d
+$COMPOSE_CMD up -d
+
+export CARDANO_NODE_SOCKET_PATH="${CLUSTER_WORKDIR}/node.socket"
 
 echo -n 'Waiting for the node socket ..'
-while ! [ -S devnet/node.socket ]
+while ! [ -S $CARDANO_NODE_SOCKET_PATH ]
 do
   echo -n "."
   sleep 0.1
 done
 echo '. done'
 
-echo "Setup cardano-cli env to connect to started network"
-export CARDANO_NODE_SOCKET_PATH=./devnet/node.socket
+echo "Changing node.socket owner to current user"
+sudo chown $USER:$USER "$CARDANO_NODE_SOCKET_PATH"
+
+echo 'Waiting for cardano node sync ..'
+
+SYNC_PROGRESS=""
+while ! [ "$SYNC_PROGRESS" = '"100.00"' ]
+do
+  SYNC_PROGRESS="$(cardano-cli query tip --testnet-magic $TESTNET_MAGIC | jq '.syncProgress')"
```

Comment:
This will not work when running the demo from docker, as you can't query the node from outside a docker container (on osx).

Reply:
>This will not work when running the demo from docker, as you can't query the node from outside a docker container (on osx).

You can run `cardano-cli` inside `cardano-node` I believe. While I forget to do that.